plugin-infra/go-plugin-infra/src/com/thoughtworks/go/plugin/infra/commons/PluginsZip.java

```diff
@@ -44,6 +44,7 @@ public PluginsZip(SystemEnvironment systemEnvironment, ZipUtil zipUtil) {
         bundledPlugins = new File(systemEnvironment.get(SystemEnvironment.PLUGIN_GO_PROVIDED_PATH));
         externalPlugins = new File(systemEnvironment.get(SystemEnvironment.PLUGIN_EXTERNAL_PROVIDED_PATH));
         this.zipUtil = zipUtil;
+        checkFilesAccessibility(bundledPlugins, externalPlugins);
```

Comment:
We prefer the constructors to not blow up with unexpected errors. Would you mind moving this check into the `create()` method?